content/Hardware Support/Generic/Reset-the-flash-memory-on-STM32H747-based-devices.md

```diff
@@ -0,0 +1,43 @@
+---
+title: "Reset the flash memory on STM32H747-based devices"
+---
+
+Learn how to reset the QSPI flash memory for STM32H747-based Arduino devices by uploading a utility sketch. The sketch will remove any previous partitioning scheme, and recreate the factory-defined partition scheme. Finally, the latest connectivity firmware and TLS certificates are reinstalled.
+
+This procedure can help resolve issues with storage or connectivity.
+
+Supported devices:
+
+* GIGA R1 WiFi
+* Opta
+* Portenta H7
+* Portenta Machine Control
+
+---
+
+**Follow these steps:**
+
+1. Download the following sketch:
+
+   [memory_partitioning.zip](https://docs.arduino.cc/3147be35e0c40957022af51f015c3e4d/memory_partitioning.zip){.link-download}
+
+1. Open the sketch in your editor:
+
+   * Arduino IDE: Extract the .zip into your [Sketchbook](https://support.arduino.cc/hc/en-us/articles/4412950938514-Open-the-Sketchbook-folder). Ensure both `memory_partitioning.ino` and `certificates.h` are inside a folder named `memory_partitioning`.
+
+   * Cloud Editor: Open the [Sketches tab](https://app.arduino.cc/sketches) and click on the **Create** button in the top-right corner. Select **Import local file**.
```

Comment:
```suggestion
   * Cloud Editor: Open the [Sketches tab](https://app.arduino.cc/sketches) and click on the **Create** button in the top-right corner. Select **Import local file** in the dropdown menu and import `memory_partitioning.zip`.
```
Consider adding a bit more detail in the Cloud Editor instructions. An instruction to import `memory_partitioning.zip` will ensure that both `memory_partitioning.ino` and `certificates.h`are imported.

Reply:
Great point!